User/InMemoryUserRepository.php

```diff
@@ -27,15 +27,15 @@ public function __construct()
 
         $this->users[] = $user;
 
-        $passwordResetUser = new User(
+        $passwordResetUser = new UserWithPassword(
```

Comment:
There used to be... I can imagine some situations where a user wouldn't have a password. Plus I think it reads better when you can say `class Admin extends UserWithPassword`. It's just an extra confirmation